### ISSUE TITLE
E3677 'java25' does not match '^(nodejs.*|python.*)$'

### ISSUE THREAD

**Issue by dmarinuswoodwing** [aws-cloudformation/cfn-lint#4397]
### CloudFormation Lint Version

1.47.0

### What operating system are you using?

Fedora 43

### Describe the bug

It seems only nodejs and python is allowed as `Runtime` for `AWS::Lambda::Function`

### Expected behavior

Many more runtimes are allowed, see:

https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-lambda-function.html#cfn-lambda-function-runtime

### Reproduction template

```yaml
Resources:
  RotationLambda:
    Type: AWS::Lambda::Function
    Properties:
      Code: s3://bucket/code.zip
      Role: arn:aws:iam::012345678901:role/role
      Runtime: 'java25'
```
